lib/TimeZoneUtil.php

```diff
@@ -94,6 +96,10 @@ private function findTimeZone(string $tzid, Component $vcalendar = null, bool $f
             return $timezone;
         }
 
+        if (!$activeCustomizedGuesser) {
```

Comment:
Since the default is `false`, I would rather add it on demand (than remove it). Furthermore, if you can the `findTimeZone` function once with `$activeCustomizedGuesser = false`, then all the next call will not have the guesser even if you set `$activeCustomizedGuesser = true`.

What do you think of something like this?
```
    private function findTimeZone(string $tzid, Component $vcalendar = null, bool $failIfUncertain = false, bool $activeCustomizedGuesser = false): DateTimeZone
    {
        foreach ($this->timezoneFinders as $timezoneFinder) {
            $timezone = $timezoneFinder->find($tzid, $failIfUncertain);
            if (!$timezone instanceof DateTimeZone) {
                continue;
            }

            return $timezone;
        }

        if ($vcalendar) {
            // We temporary add the customized timezone guesser if needed
            $guessers = $this->timezoneGuessers;
            if ($activeCustomizedGuesser) {
                $guessers[] = new GuessFromCustomizedTimeZone();
            }

            // If that didn't work, we will scan VTIMEZONE objects
            foreach ($vcalendar->select('VTIMEZONE') as $vtimezone) {
                if ((string) $vtimezone->TZID === $tzid) {
                    foreach ($guessers as $timezoneGuesser) {
                        $timezone = $timezoneGuesser->guess($vtimezone, $failIfUncertain);
                        if (!$timezone instanceof DateTimeZone) {
                            continue;
                        }

                        return $timezone;
                    }
                }
            }
        }

        if ($failIfUncertain) {
            throw new InvalidArgumentException('We were unable to determine the correct PHP timezone for tzid: '.$tzid);
        }

        // If we got all the way here, we default to whatever has been set as the PHP default timezone.
        return new DateTimeZone(date_default_timezone_get());
    }
```

The idea is that we only create and add the customizer guesser when needed and a call of `findTimeZone` cannot impact the next one

Reply:
Yes you are right, for the next call we will not have the guesser, changed!